js/ui/animation.js

```diff
@@ -75,6 +84,23 @@ var Animation = new Lang.Class({
 
     _onDestroy: function() {
         this.stop();
+    },
+
+    setFrameTimeouts: function(timeoutList) {
+        if (timeoutList.length == 0) {
```

Comment:
Perhaps the other thing to check for is that the length of the list is the same as the number of frames in the animation here?

Reply:
That's not necessary because we're wrapping if the number of frames is higher. This may be useful if we have a repeated pattern of timeouts for the frames.

Reply:
OK, let's leave it as it is.